app/main.py

```diff
@@ -1 +1,52 @@
-# write your code here
+
+class SoftwareEngineer:
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills.extend(["JavaScript", "HTML", "CSS"])
```

Comment:
The reason you're directly setting self.skills to ["JavaScript", "HTML", "CSS"] in each child class (like FrontendDeveloper) instead of using self.skills.extend([...]) is primarily due to how inheritance works and the initialization of object attributes. Here's why:

In the parent class SoftwareEngineer, self.skills is initialized as an empty list:
self.skills = []

When you create an instance of FrontendDeveloper, you want to initialize it with a specific set of skills (["JavaScript", "HTML", "CSS"]). If you use self.skills.extend([...]), you are appending skills to an already initialized self.skills list. This approach might seem fine, but in certain cases, it could lead to unexpected behavior.

Reply:
I understand about MRO and about inherited skills, but in the example don't say anything about FullStackDeveloper skills , just have an example with they skills, and i don't known, we need inherited skills from Beckend and Frontend classes or override skills?
in the first time i do this:
```
    def __init__(self, name: str) -> None:
        super().__init__(name)
        self.skills = [
            "Python",
            "SQL",
            "Django",
            "JavaScript",
            "HTML",
            "CSS",
        ]
```
but Natalia made a reject
i make it through ....extend.... and also have a reject
so, what i am doing wrong?

if i doing smth like that:... i extended Backend skills, and if i understand, i haven't do it
```
class FullStackDeveloper(
    BackendDeveloper,
    FrontendDeveloper,
):
    def __init__(self, name: str) -> None:
        super().__init__(name)
        self.skills.extend([
            "JavaScript",
            "HTML",
            "CSS"
        ])
```

Nothing about init skills:       and my first code isn't correct, and we shouldn't extend parent classes skills, why?
![image](https://github.com/user-attachments/assets/0cb5d985-f2e3-4e28-84d0-365cc9dcdc50)

Reply:
Hello, you’re right that it’s not directly asked in the task, but in general, for a full-stack developer, it is assumed that there’s no need to explicitly list of skills, as it should be a union of both backend and frontend skills.